### PR TITLE
*: support count in range query

### PR DESCRIPTION
**Pull request by xiang90**
/cc @heyitsanthony

### CODE REVIEW

File: mvcc/kv.go [hunk 1]
```diff
@@ -37,7 +49,7 @@ type KV interface {
 	// If `end` is not nil and empty, it gets the keys greater than or equal to key.
 	// Limit limits the number of keys returned.
 	// If the required rev is compacted, ErrCompacted will be returned.
-	Range(key, end []byte, limit, rangeRev int64) (kvs []mvccpb.KeyValue, rev int64, err error)
+	Range(key, end []byte, ro RangeOptions) (r *RangeResult, err error)
```
Comment (heyitsanthony) [inline mvcc/kv.go]: maybe this should be `Range(key []byte, ro RangeOptions)` since it's so similar to the client interface?

**Comment by heyitsanthony** [comment]
lgtm

File: mvcc/kv.go (hunk 1 above)
Comment (xiang90) [inline mvcc/kv.go]: the client interface is get not range. so it is reasonable to make endkey as option. here the interface is range, which should take both key and end. We can rename this to get too if we want. but i feel range is ok here.

Comment (heyitsanthony) [inline mvcc/kv.go]: ok, range is fine

**Pull request merged**